### PR TITLE
add light validation sync test

### PR DESCRIPTION
**Pull request by huangminghuang**
## Change Description
Add an integration test for messaging syncing between a production node and a light validation node.

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: tests/light_validation_sync_test.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+from Node import Node
+from Node import ReturnType
+from TestHelper import TestHelper
+from TestHelper import AppArgs
+import json
+
+###############################################################
+# nodeos_chainbase_allocation_test
+#
+# Test snapshot creation and restarting from snapshot
```
Comment (heifner) [inline tests/light_validation_sync_test.py]: Update comment for this test.

File: tests/light_validation_sync_test.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    # - account_object (bootstrap)
+    # - code_object (bootstrap)
+    # - generated_transaction_object
+    # - global_property_object
+    # - key_value_object (bootstrap)
+    # - protocol_state_object (bootstrap)
+    # - permission_object (bootstrap)
+    # The bootstrap process has created account_object and code_object (by uploading the bios contract),
+    # key_value_object (token creation), protocol_state_object (preactivation feature), and permission_object
+    # (automatically taken care by the automatically generated eosio account)
```
Comment (heifner) [inline tests/light_validation_sync_test.py]: This comment has nothing to do with this test.

File: tests/light_validation_sync_test.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    trx = {
+        "actions": [{"account": "payloadless", "name": "doit", "authorization": [{
+          "actor": "payloadless", "permission": "active"}], "data": ""}],
+        "context_free_actions": [{"account": "payloadless", "name": "doit", "data": ""}],
+        "context_free_data": ["a1b2c3", "1a2b3c"],
+    } 
+
+    cmd = "push transaction '{}' -p payloadless".format(json.dumps(trx))
+    trans = producerNode.processCleosCmd(cmd, cmd, silentErrors=False)
+    assert trans
```
Comment (heifner) [inline tests/light_validation_sync_test.py]: Other python tests use `errorExit` shouldn't that be used instead of `assert`?

Comment (huangminghuang) [reply tests/light_validation_sync_test.py]: I followed the example from node_chainbase_allocation_test.py which also uses assert. If you like more message to the assertion, I can use the expression "assert condition, message".  The problem of using errorExit() is that you need to use explicit if statement to check the condition which make it more verbose.

Comment (heifner) [reply tests/light_validation_sync_test.py]: A message to the assert I think would be helpful.

**Pull request merged**